clients/drcachesim/tests/invariant_checker_test.cpp

```diff
@@ -135,25 +147,33 @@ run_checker(const std::vector<memref_t> &memrefs, bool expect_error,
             serial_schedule_file->clear();
             serial_schedule_file->seekg(0, std::ios::beg);
         }
-        checker_no_abort_t checker(true /*offline*/, serial_schedule_file);
-        void *shard1 = checker.parallel_shard_init(1, NULL);
-        void *shard2 = checker.parallel_shard_init(2, NULL);
-        void *shard3 = checker.parallel_shard_init(3, NULL);
+        checker_no_abort_t checker(/*offline=*/true, /*serial=*/false,
+                                   serial_schedule_file);
+        void *shardA = nullptr, *shardB = nullptr, *shardC = nullptr;
         for (const auto &memref : memrefs) {
-            if (memref.instr.tid == 1)
-                checker.parallel_shard_memref(shard1, memref);
-            else if (memref.instr.tid == 2)
-                checker.parallel_shard_memref(shard2, memref);
-            else if (memref.instr.tid == 3)
-                checker.parallel_shard_memref(shard3, memref);
-            else {
+            if (memref.instr.tid == TID_A) {
```

Comment:
Wouldn't a switch-case statement easier to read than the nested if-else-if here?

Reply:
Done.